chirp/drivers/retevis_ra25.py

```diff
@@ -0,0 +1,1098 @@
+        rs = RadioSetting("compander", "Compander", rstype)
+        mem.extra.append(rs)
+
+        rstype = RadioSettingValueBoolean(_mem.nc)
+        rs = RadioSetting("nc", "Noise Cancelation", rstype)
+        mem.extra.append(rs)
+
+        return mem
+
+    # Called when a user edits UI mem
+
+    def set_memory(self, mem):
+        _mem = self._memobj.memory[mem.number - 1]
+        _mem_chans_scanned = self._memobj.chans_scanned
+        _mem_chans_present = self._memobj.chans_present
+
+        if mem.empty:
+            _mem_field = _mem_chans_present.chans_present_1 if mem.number - \
+                1 <= 256 else _mem_chans_present.chans_present_2
+            self.set_chan_status(_mem_field, mem.number -
+                                 1, 0)  # deactivate channel
+            _mem_field = _mem_chans_scanned.chans_scan_1 if mem.number - \
+                1 <= 256 else _mem_chans_scanned.chans_scan_2
+            self.set_chan_status(_mem_field, mem.number -
+                                 1, 0)  # skip scanning
+            _mem.fill_raw(b'\xFF')  # clear memory
+            return
+
+        _mem.freq = mem.freq / 10
+        _mem.name = mem.name.ljust(8)[:8]
+        _mem.txpower = self.POWER_LEVELS.index(mem.power) if mem.power else 0
+        _mem.channel_width = 0 if mem.mode == "FM" else 1
+        self.set_tones__mem(_mem, mem)
+        self.set_duplex__mem(_mem, mem)
+        self.set_offset__mem(_mem, mem)
+        _mem.step = self.VALID_TUNING_STEPS.index(mem.tuning_step)
+        # extra settings
+        for setting in mem.extra:
+            setattr(_mem, setting.get_name(), setting.value)
+
+        _mem_field = _mem_chans_scanned.chans_scan_1 if mem.number - \
+            1 <= 256 else _mem_chans_scanned.chans_scan_2
+        # scan when set(1) skip when off(0)
+        _value = 0 if mem.skip == 'S' else 1
+        self.set_chan_status(_mem_field, mem.number - 1, _value)
+
+        _mem_field = _mem_chans_present.chans_present_1 if mem.number - \
+            1 <= 256 else _mem_chans_present.chans_present_2
+        if not mem.empty:
+            self.set_chan_status(_mem_field, mem.number - 1, 1)
+
+    def get_chan_status(self, _mem_field, chan_n):
+        # checks 32 byte array
+        # used for chan present and chan skip
+        byte_num = int(chan_n // 8)
+        bit_pos = chan_n - (byte_num * 8)
+        res = 1 if (_mem_field[byte_num] & 1 << bit_pos) else 0
+        return res
+
+    def set_chan_status(self, _mem_field, chan_n, value):
+        byte_num = int(chan_n // 8)
+        bit_pos = chan_n - (byte_num * 8)
+        if value == 1:
+            _mem_field[byte_num] |= (1 << bit_pos)
+        else:
+            _mem_field[byte_num] &= ~(1 << bit_pos)
+
+    def get_offset(self, _mem, mem):
+        # gets offset from _mem for ui
+        _hex = hex(_mem.tx_offset)[2:].zfill(8)
+        str_flt = _hex[:3] + '.' + _hex[3:]
+        mem.offset = float(str_flt) * 1000000
```

Comment:
I'm not sure what's going on here, but this sort of converting back and forth between strings and int/float is not ideal. This looks to me like `bbcd` or `lbcd` types would be what you're looking for...